src/ID3Writer.mjs

```diff
@@ -18,7 +18,12 @@ import {
 
 export default class ID3Writer {
   _setIntegerFrame(name, value) {
-    const integer = parseInt(value, 10);
+    let integer;
+    if (name === 'TDAT') {
+      integer = String(parseInt(value, 10)).padStart(4, '0');
+    } else {
+      integer = parseInt(value, 10);
+    }
```

Comment:
![9233-hide-the-pain-harold](https://user-images.githubusercontent.com/2605831/228800211-26b91845-dc32-4303-b99d-fd2f1ba23a2b.png)